src/ScottPlot/Validate.cs

```diff
@@ -29,6 +29,22 @@ public static void AssertAllReal(string label, double[] values)
                     throw new ArgumentException($"{label} index {i} is invalid ({values[i]})");
         }
 
+        /// <summary>
+        /// Throw an exception if the array is null or contains NaN or infinity
+        /// </summary>
+        public static void AssertAllReal<T>(string label, T[] values)
+        {
+            if (values is null)
+                throw new ArgumentException($"{label} must not be null");
+
+            for (int i = 0; i < values.Length; i++)
+            {
+                double val = Convert.ToDouble(values[i]);
```

Comment:
This translation is very controversial. There are only 2 types of floating point numbers (double, float).
It is more correct to organize switching by type.

Reply:
Is something like this what you are suggesting?
```cs
public static void AssertAllReal<T>(string label, T[] values)
{
    if (typeof(T) == typeof(double))
        AssertAllReal(label, (double[])(object)values);
    else if (typeof(T) == typeof(float))
        AssertAllReal(label, (float[])(object)values);
    else
        throw new ArgumentException("values must be float[] or double[]");
}
```